.cirrus.yml

```
@@ -49,7 +55,12 @@ debian_default_toolchain_task:
     - make -j$(nproc) -sk
   tests_script:
     - VALGRIND_OPTS="--errors-for-leak-kinds=definite" make -j$(nproc) -sk check || (cat ./test-suite.log && false)
-    - /checks/check-built-plugins.sh
+    - |
+      if [[ -e ".expected-plugins/$CONTAINER_IMAGE" ]]; then
+        /checks/check-built-plugins.sh ".expected-plugins/$CONTAINER_IMAGE"
+      else
+        /checks/check-built-plugins.sh
+      fi
```

Would it make sense for this logic to be instead part of `check-built-plugins.sh`?
```suggestion
    - /checks/check-built-plugins.sh ".expected-plugins/$CONTAINER_IMAGE"
```

---

This is backwards compatibility behavior. If called without an argument, `check-built-plugins.sh` will fall back on the environment variable set in the Docker containers. Once the transition to a file based plugin list is complete, we can get rid of this.